core/domain/exp_services.py

```diff
@@ -1982,6 +1983,36 @@ def create_or_update_draft(
     exp_user_data.put()
 
 
+def try_upgrading_draft_to_exp_version(exp_id, user_id):
+    """Try upgrading draft change list to match the latest exploration version.
+
+    If successful (all changes successfully upgraded), we commit the upgraded
+    draft change list to the ExplorationuserDataModel and increment
+    draft_change_list_exp_version. If not, we do not make any changes to the
+    ExplorationuserDataModel.
+
+    Returns:
+        bool: True if successfully upgraded draft version.
+    """
+    exp_user_data = user_models.ExplorationUserDataModel.get(user_id, exp_id)
+    exploration = get_exploration_by_id(exp_id)
+    logging.info(
+        'Trying upgrading draft from version %s to version %s.' %
+        (exp_user_data.draft_change_list_exp_version, exploration.version))
+    new_draft_change_list = (
+        draft_upgrade_services.try_upgrade_from_version_to_version(
+            exp_user_data.draft_change_list,
+            exp_user_data.draft_change_list_exp_version, exploration.version,
+            exp_id))
+    if new_draft_change_list:
```

Comment:
In general, use "if X is not None" if what you're really checking is "non-null", as opposed to "truthy".

Reply:
Done.